chore: Add `CONTRIBUTING.md`. (#141)
mozilla-ai/any-llm: CONTRIBUTING.md

# Contributing to mozilla.ai any-llm

Thank you for your interest in contributing to this repository! This project supports the mozilla.ai goal of empowering choice and transparency when it comes to LLM usage and selection.

We welcome all kinds of contributions, from improving customization, to extending capabilities, to fixing bugs. Whether you’re an experienced developer or just starting out, your support is highly appreciated.

## **Guidelines for Contributions**

**Install**

We recommend to use [uv](https://docs.astral.sh/uv/getting-started/installation/):

```
uv venv
source .venv/bin/activate
uv sync --dev --extra all
```

**Lint**

Ensure all the checks pass:

```bash

> pre-commit run --all-files

Reviewer: This is what I use 🤷 
```suggestion
uv run pre-commit run --all-files --verbose
```

Reviewer: totally. I think this makes sure that we use the local install of pre-commit even if there is one installed globally? This way someone can run the pre-commit even if they didn't source the env yet.

```

> 

Reviewer 2: @daavoo It might just be that I don't know how to read. Good catch

> 

Reviewer: +1 on this. There is still something messed up with how we run ruff, but I haven't prioritized figuring out exactly what the issue is. Any ideas about what the problem is are welcome 🏅

**Tests**

Test changes locally to ensure functionality.

> ```bash
> pytest -v tests
> ```

Reviewer: ```suggestion
```bash
pytest -v tests/unit
pytest -v tests/integration -n auto
```
```

# Contributing to mozilla.ai any-llm

> Thank you for your interest in contributing to this repository! This project supports the mozilla.ai goal of empowering developers to integrate AI capabilities into their projects using open-source tools and models.

Reviewer: ```suggestion
Thank you for your interest in contributing to this repository! This project supports the mozilla.ai goal of empowering choice and transparency when it comes to LLM usage and selection.
```

We welcome all kinds of contributions, from improving customization, to extending capabilities, to fixing bugs. Whether you’re an experienced developer or just starting out, your support is highly appreciated.

---

## **How to Contribute**

> ### **Customize for your use-case or Extend It** 🔧

Author: This is copy-pasta from I am not sure where. IMHO, we could drop it and keep this doc focused on instructions for install,lint,test,docs but I see most repos having this generic test that feels a little bit like LLM generated

- Fork this repo and customize it for your own use-case or even extend its capabilities.
- We'd love to see what you've built!

### **Browse Existing Issues** 🔍
- Check the Issues page to see if there are any tasks you'd like to tackle.
- Look for issues labeled **`good first issue`** if you're new to the project—they're a great place to start.

### **Report Issues** 🐛
- Found a bug? Open a Bug Report by clicking on 'New Issue'
- Provide as much detail as possible, including the steps to reproduce the issue and Expected vs. actual behavior

### **Suggest Features** 🚀
- Have an idea for improving the project? Open a Feature Request by clicking on 'New Issue'

> - Share why the feature is important and any alternative solutions you’ve considered.

Reviewer: To me it doesn't seem like a value add? Describing how github works
```suggestion
```

### **Submit Pull Requests** 💻
- Fork the repository and create a new branch for your changes.
- Follow the [Guidelines for Contributions](#guidelines-for-contributions)
- Ensure your branch is up-to-date with the main branch before submitting the PR.
- Please follow the PR template, adding as much detail as possible, including how to test the changes

---

### **Guidelines for Contributions**

**Install**

We recommend to use [uv](https://docs.astral.sh/uv/getting-started/installation/):

```
uv venv
source .venv/bin/activate

> uv sync --dev --extra all

Reviewer: This is what I use 🤷 
```suggestion
uv sync --all-extras -U --python=3.13
```

Author: Why not `--dev`? (installs lint, test and docs deps, which is relevant for local development)

And why `--python=3.13` in the `sync` instead of when creating the `uv venv`?

Reviewer: When testing different python versions and the venv already existed, `--python` swapped the version for me on the fly, IIRC.

the `--all-extras` tag installs everything, both the `all` extra as well as everything else (which includes the dev deps). Not sure if that was your question
